### DISCUSSION TITLE
Logging: Remove Color Message Extra Key

### DISCUSSION THREAD

**Discussion by josheppinette** [Kludex/uvicorn#1377] [Q&A]
I am adding a work around in my custom [logfmt formatter](https://github.com/jteppinette/python-logfmter/issues/1) to exclude keys due to the `color_message` key being passed through to the custom color logging formatter.

I don't really see this as a real issue, but if there was another way to solve this problem, then anyone with a custom log formatter wouldn't receive this unnecessary key.

Here is my current log output without the customization to my logging formatter:
```
at=INFO msg="Started server process [97819]" color_message="Started server process [%d]"
at=INFO msg="Waiting for application startup."
at=INFO msg="Application startup complete."
at=INFO msg="Uvicorn running on http://127.0.0.1:8000 (Press CTRL+C to quit)" color_message="Uvicorn running on %s://%s:%d (Press CTRL+C to quit)"
```

**Comment by rminderhoud** [comment]
For anyone stumbling upon this:

- If using custom formatter, you'll have to drop the extra keys manually since uvicorn passes it to the log messages under the `extras` kwarg
- If using `structlog` don't use the `structlog.stdlib.ExtraAdder()` processor
